pybuildkite/client.py

```diff
@@ -50,13 +50,14 @@ def request(self, method, url, query_params=None, body=None, headers=None):
             body = self._clean_query_params(body)
 
         response = requests.request(method, url, headers=headers, params=query_params, json=body)
-        response.raise_for_status()
-
-        # TODO what should be returned
-        if headers:
+        try:
+            response.raise_for_status()
+        except:
             return response.text
-
-        return response.json()
+        if response.status_code == 204:
```

Comment:
is the `json` object null on a DELETE request? If so I'd rather return `response.ok` based on us doing DELETE and it being a "success" (any 200 code), rather than specifically 204

Reply:
The success response for DELETE request is 204 with no content, so response.json would return an error.